src/aks-preview/azext_aks_preview/_roleassignments.py

```diff
@@ -0,0 +1,136 @@
+    return list(graph_client.objects.get_objects_by_object_ids(params))
+
+
+def resolve_object_id(cli_ctx, assignee):
+    client = get_graph_rbac_management_client(cli_ctx)
+    result = None
+    if assignee.find('@') >= 0:  # looks like a user principal name
+        result = list(client.users.list(
+            filter="userPrincipalName eq '{}'".format(assignee)))
+    if not result:
+        result = list(client.service_principals.list(
+            filter="servicePrincipalNames/any(c:c eq '{}')".format(assignee)))
+    if not result:  # assume an object id, let us verify it
+        result = _get_object_stubs(client, [assignee])
+
+    # 2+ matches should never happen, so we only check 'no match' here
+    if not result:
+        raise CLIError(
+            "No matches in graph database for '{}'".format(assignee))
+
+    return result[0].object_id
+
+
+def resolve_role_id(role, scope, definitions_client):
+    role_id = None
+    try:
+        uuid.UUID(role)
+        role_id = role
+    except ValueError:
+        pass
+    if not role_id:  # retrieve role id
+        role_defs = list(definitions_client.list(
+            scope, "roleName eq '{}'".format(role)))
+        if not role_defs:
+            raise CLIError("Role '{}' doesn't exist.".format(role))
+        if len(role_defs) > 1:
+            ids = [r.id for r in role_defs]
+            err = "More than one role matches the given name '{}'. Please pick a value from '{}'"
+            raise CLIError(err.format(role, ids))
+        role_id = role_defs[0].id
+    return role_id
+
+
+def build_role_scope(resource_group_name, scope, subscription_id):
+    subscription_scope = '/subscriptions/' + subscription_id
+    if scope:
+        if resource_group_name:
+            err = 'Resource group "{}" is redundant because scope is supplied'
+            raise CLIError(err.format(resource_group_name))
+    elif resource_group_name:
+        scope = subscription_scope + '/resourceGroups/' + resource_group_name
+    else:
+        scope = subscription_scope
+    return scope
+
+
+def create_role_assignment(cli_ctx, role, assignee,
+                           is_service_principal=True, resource_group_name=None, scope=None, resolve_assignee=True):
+    return _create_role_assignment(cli_ctx,
+                                   role, assignee, resource_group_name,
+                                   scope, resolve_assignee=(is_service_principal and resolve_assignee))
+
+
+def _create_role_assignment(cli_ctx, role, assignee,
+                            resource_group_name=None, scope=None, resolve_assignee=True):
+    from azure.cli.core.profiles import ResourceType, get_sdk
+    factory = get_auth_management_client(cli_ctx, scope)
+    assignments_client = factory.role_assignments
+    definitions_client = factory.role_definitions
+
+    scope = build_role_scope(
+        resource_group_name, scope, assignments_client.config.subscription_id)

```

Comment:
better to check whether assignments_client.config is None or not before use it.

Reply:
Hi Andy, a lot of this code is not mine. It was simply moved around for organization purposes and to avoid circular dependencies. I ensured everything still works, but did not change any code functionality except for functions that were directly related to addons (and there were only 2 of them). I'm not sure it would be safe for me to change the functionality of these other functions without knowing there purpose.

This one seems easy enough to add in, however, if you want me to.